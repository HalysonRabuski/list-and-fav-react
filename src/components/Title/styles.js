import styled from "styled-components";
import { Typography } from "antd";

const { Title, Text } = Typography;

export const StyledTitle = styled(Text)`
  &.ant-typography {
    color: #f5f5f5;
    line-height: normal !important;
    height: 64px;
    line-height: 64px !important;
    margin: 0;
    font-size: 2rem;
  }
`;
