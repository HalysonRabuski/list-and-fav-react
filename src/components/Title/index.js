import React from "react";
import { StyledTitle } from "./styles";

function Title(props) {
  return <StyledTitle data-testid="title">{props.text}</StyledTitle>;
}

export default Title;
