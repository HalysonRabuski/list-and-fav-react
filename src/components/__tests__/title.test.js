import { render, screen, cleanup } from "@testing-library/react";
import renderer from "react-test-renderer";
import Title from "../Title";

afterEach(() => {
  cleanup();
});

test("should render title component", () => {
  render(<Title text="test" />);
  const titleElement = screen.getByTestId("title");
  expect(titleElement).toBeInTheDocument();
  expect(titleElement).toHaveTextContent("test");
});

test("should render title component like snapshot", () => {
  const titleElement = renderer.create(<Title />).toJSON();
  expect(titleElement).toMatchSnapshot();
});
