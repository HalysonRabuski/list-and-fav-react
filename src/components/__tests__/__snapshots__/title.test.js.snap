// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render title component like snapshot 1`] = `
<span
  className="ant-typography sc-bdvvaa goBTMR"
  data-testid="title"
  style={
    Object {
      "WebkitLineClamp": undefined,
    }
  }
>
  
</span>
`;
